File: Baekjoon/1182.cpp
#include <iostream>
using namespace std;

int n, s, cnt;
int num[20];
bool chk[20];

void solve(int idx, int sum) {	
	if (idx == n) {
		if (sum == s)
			cnt++;
		return;
	}

	solve(idx + 1, sum + num[idx]);
	solve(idx + 1, sum);

}

int main() {
	cin >> n >> s;
	
	for (int i = 0; i < n; i++) {
		cin >> num[i];
	}
	solve(0, 0);

	// 공집합인 경우 제외
	if (s == 0)
		cnt--;

	cout << cnt << '\n';

	return 0;
}

File: Baekjoon/1182.kt
var ans = 0

fun dfs(list: List<Int>, idx: Int, sum: Int, s: Int, cnt: Int) {
    if (idx == list.size) {
        if (sum == s && cnt > 0) ans++
        return
    }

    dfs(list, idx + 1, sum, s, cnt)
    dfs(list, idx + 1, sum + list[idx], s, cnt + 1)
}

fun main() {
    val (n, s) = readLine()!!.split(" ").map { it.toInt() }
    val list = readLine()!!.split(" ").map { it.toInt() }
    dfs(list, 0, 0, s, 0)
    println(ans)
}